stream.go

```diff
@@ -0,0 +1,61 @@
+package manta
+
+import (
+	"io"
+
+	"github.com/dotabuff/manta/dota"
+)
+
+// stream wraps an io.Reader to provide functions necessary for reading the
+// outer replay structure.
+type stream struct {
+	r io.Reader
+}
+
+// newStream creates a new stream from a given io.Reader
+func newStream(r io.Reader) *stream {
+	return &stream{r}
+}
+
+// readBytes reads the given number of bytes from the reader
+func (s *stream) readBytes(n uint32) ([]byte, error) {
+	buf := make([]byte, n)
```

Comment:
if you want to be alloc-conscious, you can create a single buffer up front when `stream` is assembled and always operate on the same buffer, recycling it through the course of the program. `encoding/json` does this internally with the json encoder. Recycling a single buffer here could save up to N allocations, where N is the number of times you call `readBytes`. I'm not sure how many that is, or if it's ever done in a tight loop. If N is small, don't bother.